src/Microsoft.Extensions.Logging.Testing/AssemblyTestLog.cs

```diff
@@ -234,6 +252,17 @@ private static SerilogLoggerProvider ConfigureFileLogging(string fileName)
             return new SerilogLoggerProvider(serilogger, dispose: true);
         }
 
+        private static string RemoveIllegalFileChars(string s)
+        {
+            var sb = new StringBuilder();
+
+            foreach (var c in s)
+            {
+                sb.Append(InvalidFileChars.Contains(c) ? Convert.ToByte(c).ToString("x") : $"{c}");
```

Comment:
Not when we have space in the invalid character list, please, let's not use hex, a bunch of random lower case characters would make file names completely unreadable.

Reply:
Only append underscore if the previous character is not underscore. `A_B` is better than `A_____B` and might help long paths.